Patch written by Kurt Lust, UAntwerp
It deals with a routine specific to the Intel compiler that has not been properly adapted
when the definition of cpu_detect changed from 2.6 to 2.8. It has the likely side effect
of disabling AVX512 as that routine does not seem to be properly adapted to AVX512.
The bug was already posted on the x265 forum, so hopefully 2.8.1 will properly correct it.
--- x265_2.8/source/common/x86/asm-primitives.cpp.orig  2018-07-19 16:18:46.847239000 +0200
+++ x265_2.8/source/common/x86/asm-primitives.cpp       2018-07-19 16:20:09.757634575 +0200
@@ -5402,7 +5402,7 @@
 // CPU dispatcher function
 void PFX(intel_cpu_indicator_init)(void)
 {
-    uint32_t cpu = x265::cpu_detect();
+    uint32_t cpu = x265::cpu_detect(false);

     if (cpu & X265_CPU_AVX)
         __intel_cpu_indicator = 0x20000;
